File: src/boinc-6.10.58/test/test_masterurl_failure.php
#!/usr/local/bin/php -q
<?php {
    // $Id: test_masterurl_failure.php 1510 2003-06-17 01:36:47Z quarl $

    $use_proxy_html = 1;

    include_once("test_uc.inc");
    test_msg("master url failure");

    $project = new ProjectUC;

    start_proxy('exit 1 if $nconnections < 4; if_done_kill(); if_done_ping();');

    // TODO: verify it took ??? seconds

    // TODO: time out after ??? seconds and fail this test

    $project->start_servers_and_host();
    $project->validate_all_and_stop();

    test_done();
} ?>

File: src/boinc-7.0.36/test/test_masterurl_failure.py
#!/usr/bin/env python

## $Id: test_masterurl_failure.py 1655 2003-07-10 00:35:51Z quarl $

from test_uc import *

if __name__ == '__main__':
    test_msg("scheduler exponential backoff (master url failure)")
    Proxy('close_connection if $nconnections < 2; ', html=1)
    # Proxy(('print "hi nc=$nconnections start=$start nchars=$nchars url=$url\n"; ' +
    #        'close_connection if $nconnections < 2; '),
    #       html=1)
    # Proxy(( 'print "hi nc=$nconnections start=$start nchars=$nchars url=$url\n"; ' +
    #        ''),
    #       html=1)
    ProjectUC(short_name='test_masterurl_failure')
    run_check_all()

    ## TODO: verify it took ??? seconds
